build/common.ps1

```diff
@@ -532,7 +533,7 @@ Function Invoke-ILMerge {
     Trace-Log 'Creating the ilmerged nuget.exe'
     $opts = , 'NuGet.exe'
     $opts += $buildArtifacts
-    $opts += "/out:$Artifacts\NuGet.exe"
+    $opts += "/out:$NuGetExeOutDev14\NuGet.exe"
```

Comment:
Made this and other changes (mainly propagated csproj changes to test projects). Please review in following commit. Will squash these two.